src/redux/modules/volunteer.ts

```diff
@@ -25,17 +51,40 @@ const setVolunteers = (volunteers: Volunteer[]): VolunteerActionTypes => {
   };
 };
 
-export const selectVolunteer = (volunteer: Volunteer): VolunteerActionTypes => {
+export const loading = (): VolunteerActionTypes => {
+  return {
+    type: LOADING,
+    payload: null,
+  };
+};
+
+export const handleError = (error: ErrorResponse): VolunteerActionTypes => {
+  return {
+    type: ERROR,
+    payload: error,
+  };
+};
+
+const setSelectedVolunteer = (volunteer: Volunteer): VolunteerActionTypes => {
   return {
     type: SELECT_VOLUNTEER,
     payload: volunteer,
   };
 };
 
+const loadingDetails = (): VolunteerActionTypes => {
+  return {
+    type: LOADING_DETAILS,
+    payload: null,
+  };
+};
+
 // Reducer
 const initialState: VolunteerState = {
   all_volunteers: [],
   loading: false,
+  error: {} as ErrorResponse,
+  loading_details: false,
   selected_volunteer: {} as Volunteer,
```

Comment:
oh wow isnt this breaking? we should probably set it to null instead of {} right?
I think {}